### PR TITLE
feat: Import new Fondue icons

### PR DESCRIPTION
**Pull request by thisguyscode**
- Uses `arcade-icons` as source for new icons, without removing the icons that were already generated from old source
- Changes generation script to remove some sizing and such
- Adds generation of dynamic icon, which includes all variants of the icon shape
- Refactored the icon generation in general

Essentially there should be no breaking changes, and we can slowly migrate to the new Icons

### CODE REVIEW

**Review by jonasbuechel — CHANGES_REQUESTED** [review]
It works great! 🥳  I added some suggestions regarding usage of TypeScript.

Main issue: the bundle size will be increased by >5.5 MB! As tree-shaking is not working yet in Arcade we need to wait to merge until we fixed it.
Screenshot 2022-02-22 at 17 11 39

**Comment by thisguyscode** [comment]
@jonasbuechel Thanks for the typescript tips, that's great help. Looking forward to solving tree-shaking :D

**Comment by thisguyscode** [comment]
@jonasbuechel What's the best way to apply these suggestions? Via the Github UI or should I do the manual work locally and push the changes?

**Comment by jonasbuechel** [comment]
> 

You can directly commit it via GitHub UI. Just ensure to doublecheck my code as well and have a test run locally (get the changes via pull)

**Comment by imoutaharik** [comment]
@jonasbuechel This Pr is ready but deployment is failing.
FATAL ERROR: Ineffective mark-compacts near heap limit Allocation failed - JavaScript heap out of memory

File: scripts/generateReactIcons.ts [hunk 1]
```diff
@@ -2,52 +2,107 @@
...
 import { writeFile, readFile } from "fs/promises";
+import { camelCase, toUpper } from "lodash";
 // @ts-ignore
 import { transform } from "@svgr/core";
 import { Entry } from "fast-glob/out/types";
 import { IconTemplate } from "../src/foundation/Icon/IconTemplate";
+import { IconComponent, IconTemplateDynamic } from "../src/foundation/Icon/IconTemplateDynamic";
 
-export const GENERATED_ICONS_INDEX_PATH = "src/foundation/Icon/Generated/index.ts";
+const ICON_SOURCE_DIRECTORY = `node_modules/@frontify/arcade-icons/icons/`;
```
Comment (jonasbuechel) [inline scripts/generateReactIcons.ts:12]:
```suggestion
const ICON_SOURCE_DIRECTORY = `node_modules/@frontify/fondue-icons/icons/`;
```

**Review by jonasbuechel — CHANGES_REQUESTED** [review]

File: src/foundation/Icon/Generated/IconArrowBidirectional.tsx [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import React, { memo } from "react";
+import { GeneratedIconProps } from "@foundation/Icon/IconProps";
+import { IconSize } from "@foundation/Icon/IconSize";
+
+import IconArrowBidirectional12 from './IconArrowBidirectional12';;
```
Comment (jonasbuechel) [inline src/foundation/Icon/Generated/IconArrowBidirectional.tsx:5]: There seems to be an error in the template we use for generation which produces double semicolons.
```suggestion
import IconArrowBidirectional12 from './IconArrowBidirectional12';
```

**Review by jonasbuechel — CHANGES_REQUESTED** [review]

File: src/foundation/Icon/Generated/IconArrowBidirectional.tsx (hunk 2 above)
Comment (imoutaharik) [reply src/foundation/Icon/Generated/IconArrowBidirectional.tsx]: Fixed!

**Review by imoutaharik — APPROVED** [review]

**Pull request merged**